ndc-sqlgen/src/main/kotlin/io/hasura/ndc/sqlgen/BaseGenerator.kt

```diff
@@ -41,26 +41,26 @@ sealed interface BaseGenerator {
         // unwrap single value for use in all but the IN operator
         // OR return falseCondition if listVal is empty
         val singleVal = listVal.firstOrNull() ?: return DSL.falseCondition()
+        val castedValue = castValue(singleVal, columnType)
 
         return when (operator) {
-            ApplyBinaryComparisonOperator.EQ -> col.eq(castValue(singleVal, columnType))
-            ApplyBinaryComparisonOperator.GT -> col.gt(castValue(singleVal, columnType))
-            ApplyBinaryComparisonOperator.GTE -> col.ge(castValue(singleVal, columnType))
-            ApplyBinaryComparisonOperator.LT -> col.lt(castValue(singleVal, columnType))
-            ApplyBinaryComparisonOperator.LTE -> col.le(castValue(singleVal, columnType))
+            ApplyBinaryComparisonOperator.EQ -> col.eq(castedValue)
+            ApplyBinaryComparisonOperator.GT -> col.gt(castedValue)
+            ApplyBinaryComparisonOperator.GTE -> col.ge(castedValue)
+            ApplyBinaryComparisonOperator.LT -> col.lt(castedValue)
+            ApplyBinaryComparisonOperator.LTE -> col.le(castedValue)
             ApplyBinaryComparisonOperator.IN -> col.`in`(listVal.map { castValue(it, columnType) })
             ApplyBinaryComparisonOperator.IS_NULL -> col.isNull
             ApplyBinaryComparisonOperator.LIKE -> col.like(singleVal as Field<String>)
             ApplyBinaryComparisonOperator.CONTAINS -> col.contains(singleVal as Field<String>)
         }
     }
 
-    // 
     fun castValue(value: Any, scalarType: NDCScalar?): Any {
         return when (scalarType) {
-            NDCScalar.TIMESTAMPTZ -> DSL.cast(value, java.sql.Timestamp::class.java)
-            NDCScalar.TIMESTAMP -> DSL.cast(value, java.sql.Timestamp::class.java)
-            NDCScalar.DATE -> DSL.cast(value, java.sql.Date::class.java)
+            NDCScalar.TIMESTAMPTZ -> DSL.cast(value, SQLDataType.TIMESTAMPWITHTIMEZONE)
+            NDCScalar.TIMESTAMP -> DSL.cast(value, SQLDataType.TIMESTAMP)
+            NDCScalar.DATE -> DSL.cast(value, SQLDataType.DATE)
```

Comment:
I thought when I tried this it errored?

Reply:
> Or do you think this is safe as is?

I _feel_ like explicit casts shouldn't cause anything to break on the other dialects, but if you want me to move this into an oracle-specific implementation I can certainly do that.

> I thought when I tried this it errored?

I tested it against a `DATE` column and a `TIMESTAMP` column and it seemed to work.
Only reason why I modified this was in case there is some specific behavior for `TIMESTAMPTZ` vs `TIMESTAMP`, because `java.sql` types don't have the ability to discriminate